hail/python/hail/__init__.py

```diff
@@ -1,9 +1,13 @@
 import pkg_resources
 import sys
+import pyspark
```

Comment:
I don't want to import pyspark if we don't have to. Only the Spark backend should import pyspark -- this check should go there, and not in code run at import-time.